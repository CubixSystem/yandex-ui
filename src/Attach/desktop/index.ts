export * from '../Attach@desktop';
export * from '../_size/Attach_size_l';
export * from '../_size/Attach_size_m';
export * from '../_size/Attach_size_s';
