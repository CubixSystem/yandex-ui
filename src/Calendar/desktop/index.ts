// polyfills
import '../../polyfills/pointerfocus';
// Public API (desktop)
export { useCalendarState } from '../hooks/useCalendarState';

export { cnCalendar, Calendar as unstable_Calendar, DateValue, CalendarProps, CalendarType } from '../Calendar';
export { Day as DayCalendar } from '../Day';
export { Year as YearCalendar } from '../Year';
export { Month as MonthCalendar } from '../Month';

export { withSizeS } from '../_size/Calendar_size_s';
export { withSizeM } from '../_size/Calendar_size_m';
export { withSizeL } from '../_size/Calendar_size_l';
export { withViewDefault } from '../_view/Calendar_view_default';
