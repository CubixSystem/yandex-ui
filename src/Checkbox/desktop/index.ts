export * from '../Checkbox@desktop';
export * from '../_indeterminate/Checkbox_indeterminate';
export * from '../_lines/Checkbox_lines_multi';
export * from '../_lines/Checkbox_lines_one';
export * from '../_size/Checkbox_size_m';
export * from '../_size/Checkbox_size_s';
export * from '../_theme/Checkbox_theme_normal@desktop';
export * from '../_theme/Checkbox_theme_pseudo@desktop';
export * from '../_view/Checkbox_view_default@desktop';
export * from '../_view/Checkbox_view_outline@desktop';
