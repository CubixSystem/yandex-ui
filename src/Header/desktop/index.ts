export * from '../Header';
export * from '../Header.const';
export * from '../Nav/Header-Nav';
export * from '../Search/Header-Search';
export * from '../Logo/Header-Logo';
export * from '../Logo/Inline-Logo';
export * from '../Logo/Logoaas';
export * from '../Icons/Header-Icons';
export * from '../Icons/Mirror-Icon';
export * from '../Icons/Weather-Icon';
export * from '../Icons/Sport-Icon';
export * from '../Icons/News-Icon';
export * from '../Icons/Services-Icon';
export * from '../Icons/Inline-Icons';
