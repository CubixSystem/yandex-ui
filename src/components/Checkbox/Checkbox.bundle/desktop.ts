import { FC } from 'react';
import { compose, composeU } from '@bem-react/core';

// base
import { ICheckboxProps as ICheckboxDesktopProps, Checkbox as CheckboxDesktop } from '../Checkbox@desktop';
// _lines
import { withLinesMulti } from '../_lines/Checkbox_lines_multi';
import { withLinesOne } from '../_lines/Checkbox_lines_one';
// _size
import { withSizeM } from '../_size/Checkbox_size_m';
import { withSizeS } from '../_size/Checkbox_size_s';
// _theme
import { withThemeNormal } from '../_theme/Checkbox_theme_normal@desktop';
import { withThemePseudo } from '../_theme/Checkbox_theme_pseudo@desktop';
// _view
import { withViewDefault } from '../_view/Checkbox_view_default@desktop';
// _indeterminate
import { withIndeterminate } from '../_indeterminate/Checkbox_indeterminate';

export * from '../Checkbox@desktop';

export interface ICheckboxProps extends ICheckboxDesktopProps {
    lines?: 'multi' | 'one';
    size?: 'm' | 's';
    theme?: 'normal' | 'pseudo';
    view?: 'default';
    indeterminate?: boolean;
}

export const Checkbox = compose(
    withViewDefault,
    withIndeterminate,
    composeU(withLinesMulti, withLinesOne),
    composeU(withSizeM, withSizeS),
    composeU(withThemeNormal, withThemePseudo),
)(CheckboxDesktop) as FC<ICheckboxProps>;
