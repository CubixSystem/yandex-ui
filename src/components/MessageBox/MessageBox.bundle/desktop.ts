import { compose, composeU } from '@bem-react/core';

import { MessageBox as MessageBoxCommon, MessageBoxPopup as MessageBoxPopupCommon } from '../MessageBox@desktop';
import { withViewDefault } from '../_view/MessageBox_view_default';
import { withViewPromo } from '../_view/MessageBox_view_promo';
import { withViewInverse } from '../_view/MessageBox_view_inverse';
import { withSizeL } from '../_size/MessageBox_size_l';
import { withSizeM } from '../_size/MessageBox_size_m';
import { withSizeS } from '../_size/MessageBox_size_s';

export * from '../MessageBox@desktop';

const enhance = compose(
    composeU(withViewDefault, withViewPromo, withViewInverse),
    composeU(withSizeL, withSizeM, withSizeS),
);

export const MessageBox = enhance(MessageBoxCommon);
export const MessageBoxPopup = enhance(MessageBoxPopupCommon);
