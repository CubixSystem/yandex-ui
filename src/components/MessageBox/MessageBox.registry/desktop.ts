import { Registry } from '@bem-react/di';
import { compose } from '@bem-react/core';

import { Popup as PopupDesktop } from '../../Popup/Popup@desktop';
import { withViewDefault } from '../../Popup/_view/Popup_view_default';
import { withTargetAnchor } from '../../Popup/_target/Popup_target_anchor';
import { withNonvisual } from '../../Popup/_nonvisual/Popup_nonvisual';
import { cnMessageBox } from '../MessageBox';
import { withOutsideClick } from '../../../hocs/withOutsideClick/withOutsideClick';
import { withZIndex } from '../../../hocs/withZIndex/withZIndex';

const Popup = compose(
    withViewDefault,
    withTargetAnchor,
    withNonvisual,
    withOutsideClick,
    withZIndex,
)(PopupDesktop);

export const registry = new Registry({ id: cnMessageBox() });

registry.set('Popup', Popup);
