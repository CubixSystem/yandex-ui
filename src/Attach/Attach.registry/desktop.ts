import { Registry } from '@bem-react/di';

import { cnAttach } from '../Attach.const';
import { Holder } from '../Holder/Attach-Holder';
import { Control } from '../Control/Attach-Control';

export const attachRegistry = new Registry({ id: cnAttach() });

attachRegistry
    .set('Holder', Holder)
    .set('Control', Control)
    .set('Button', () => {
        // TODO: Использовать error api из новой версии react.
        throw new Error(`Для работы "Attach" требуется установить компонент "Button" в реестр "${cnAttach()}".`);
    });
