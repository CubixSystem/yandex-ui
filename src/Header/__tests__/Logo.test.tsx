import React, { createRef } from 'react';

import { createClientRender, screen } from '../../internal/testing';
import { Logoaas, YandexLogo, InlineLogo } from '../desktop';

const render = createClientRender();

describe('Logoaas', () => {
    test('should render Яндекс by default', () => {
        const { container } = render(<Logoaas />);

        expect(container).toMatchSnapshot();
    });

    test('should set props', () => {
        const innerRef = createRef<HTMLDivElement>();
        const { container } = render(
            <Logoaas circle size={12} color="fff" first="00f" href="//ya.ru" innerRef={innerRef} />,
        );

        expect(container).toMatchSnapshot();
    });

    test('should set tld', () => {
        render(<Logoaas tld="com" />);

        expect(screen.getByRole('link')).toHaveAttribute('href', '//yandex.com');
    });
});

describe('YandexLogo', () => {
    test('should render Яндекс by default', () => {
        const { container } = render(<YandexLogo />);

        expect(container).toMatchSnapshot();
    });

    test('should set props', () => {
        const { container } = render(<YandexLogo circle lang="en" href="//ya.ru" />);

        expect(container).toMatchSnapshot();
    });
});

describe('InlineLogo', () => {
    test('should render src', () => {
        const { container } = render(<InlineLogo src="ya.svg" />);

        expect(container).toMatchSnapshot();
    });

    test('should set props', () => {
        const { container } = render(<InlineLogo height={2} width={1} src="//ya.ru" />);

        expect(container).toMatchSnapshot();
    });
});
