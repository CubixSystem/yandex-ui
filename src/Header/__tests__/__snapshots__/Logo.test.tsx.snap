// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLogo should render src 1`] = `
<div>
  <span
    class="Link YandexHeader-Logo-Inline"
    tabindex="-1"
  >
    <img
      class="YandexHeader-Image"
      src="ya.svg"
    />
  </span>
</div>
`;

exports[`InlineLogo should set props 1`] = `
<div>
  <span
    class="Link YandexHeader-Logo-Inline"
    tabindex="-1"
  >
    <img
      class="YandexHeader-Image"
      height="2"
      src="//ya.ru"
      width="1"
    />
  </span>
</div>
`;

exports[`Logoaas should render Яндекс by default 1`] = `
<div>
  <a
    class="Link YandexHeader-Logoaas"
    href="//yandex.ru"
    tabindex="-1"
  >
    <img
      class="YandexHeader-Image"
      src="//yastatic.net/q/logoaas/Яндекс.svg?size=32&color=000&first=f00&circle=0&single=0"
    />
  </a>
</div>
`;

exports[`Logoaas should set props 1`] = `
<div>
  <a
    class="Link YandexHeader-Logoaas"
    href="//ya.ru"
    tabindex="-1"
  >
    <img
      class="YandexHeader-Image"
      src="//yastatic.net/q/logoaas/Яндекс.svg?size=12&color=fff&first=00f&circle=1&single=0"
    />
  </a>
</div>
`;

exports[`YandexLogo should render Яндекс by default 1`] = `
<div>
  <a
    class="Link YandexHeader-Logo YandexHeader-Logo_lang_ru"
    href="//yandex.ru"
    tabindex="-1"
  />
</div>
`;

exports[`YandexLogo should set props 1`] = `
<div>
  <a
    class="Link YandexHeader-Logo YandexHeader-Logo_circle YandexHeader-Logo_lang_en"
    href="//ya.ru"
    tabindex="-1"
  />
</div>
`;
